vllm/model_executor/models/commandr.py

```diff
@@ -171,12 +171,26 @@ def __init__(
             rope_scaling=self.rope_scaling,
             is_neox_style=False,
         )
+
+        self.sliding_window = getattr(config, "sliding_window", None)
+        # Model v2 has sliding windows, v1 does not
+        self.v1 = self.sliding_window is None
+
+        layer_idx = extract_layer_index(prefix)
+        layer_has_sliding_window = (
+            getattr(config, "sliding_window_pattern", False)
+            and (layer_idx + 1) % self.config.sliding_window_pattern != 0)
+
+        if not layer_has_sliding_window:
+            self.sliding_window = None
```

Comment:
```suggestion
        sliding_window = getattr(config, "sliding_window", None)
        # Model v2 has sliding windows, v1 does not
        self.v1 = sliding_window is None

        layer_idx = extract_layer_index(prefix)
        layer_has_sliding_window = (
            getattr(config, "sliding_window_pattern", False)
            and (layer_idx + 1) % self.config.sliding_window_pattern != 0)

        if layer_has_sliding_window:
            self.sliding_window = sliding_window
```

Let's avoid assigning to `self.sliding_window` prematurely.

Reply:
Done.